From 8862c8b403570d917a31910dfe39de2beed56ceb Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Fri, 05 Nov 2021 12:47:52 +0200
Subject: [PATCH] [SW-0] gaudi2: arc: modify arc_init_dccm routine

Currently arc_init_dccm routine is used to clear ARC's DCCM memory.
This patch modifies function to make generic so that it can be
also re-used to clear ARC ICCM memory.

Change-Id: Ia08de2da21e2fc9561f7ce213716d1fa5ef44da2
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index 4404d9a..ce88ef7 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -96,7 +96,7 @@
 	regmap_writel(ARC_RST_VALUE, mmPSOC_ARC0_CFG_ARC_RST);
 
 	/* Clear DCCM memory */
-	init_arc_dccm(0);
+	init_arc_mem(0, ARC0_DCCM_FULL_BASE, ARC0_DCCM_SIZE);
 }
 
 void hl_plat_early_init(void)
diff --git a/arch/gaudi2/arm/common.c b/arch/gaudi2/arm/common.c
index 52bfafb..992dd92 100644
--- a/arch/gaudi2/arm/common.c
+++ b/arch/gaudi2/arm/common.c
@@ -171,8 +171,8 @@
 	regmap_writel(ARC_RST_VALUE, mmPSOC_ARC1_CFG_ARC_RST);
 
 	/* Clear DCCM memory */
-	init_arc_dccm(0);
-	init_arc_dccm(1);
+	init_arc_mem(0, ARC0_DCCM_FULL_BASE, ARC0_DCCM_SIZE);
+	init_arc_mem(1, ARC1_DCCM_FULL_BASE, ARC1_DCCM_SIZE);
 }
 
 void hl_plat_init(void)
diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 5c7e385..90e7cf5 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -988,26 +988,15 @@
 	regmap_writel(0x1, rst_offset);
 }
 
-#define HL_INIT_ARC_DCCM_RETRIES	50
-void init_arc_dccm(u8 id)
+#define HL_INIT_ARC_MEM_RETRIES	50
+void init_arc_mem(u8 id, u64 addr, u32 size)
 {
-	u32 retry = HL_INIT_ARC_DCCM_RETRIES;
-	u32 addr_h, addr_l, size;
+	u32 retry = HL_INIT_ARC_MEM_RETRIES;
 	u32 mem_cfg, cpy_sts;
 
-	if (id) {
-		addr_l = (u32)ARC1_DCCM_FULL_BASE;
-		addr_h = (u32)(ARC1_DCCM_FULL_BASE >> 32);
-		size = ARC1_DCCM_SIZE;
-	} else {
-		addr_l = (u32)ARC0_DCCM_FULL_BASE;
-		addr_h = (u32)(ARC0_DCCM_FULL_BASE >> 32);
-		size = ARC0_DCCM_SIZE;
-	}
-
 	/* Use memblock to initialize dccm memory */
-	regmap_writel(addr_l, mmPSOC_GLOBAL_CONF_MEM_CPY_DEST_ADDR_L);
-	regmap_writel(addr_h, mmPSOC_GLOBAL_CONF_MEM_CPY_DEST_ADDR_H);
+	regmap_writel((u32)addr, mmPSOC_GLOBAL_CONF_MEM_CPY_DEST_ADDR_L);
+	regmap_writel((u32)(addr >> 32), mmPSOC_GLOBAL_CONF_MEM_CPY_DEST_ADDR_H);
 
 	mem_cfg = regmap_readl(mmPSOC_GLOBAL_CONF_MEM_CPY_CTRL2);
 	mem_cfg &= ~(PSOC_GLOBAL_CONF_MEM_CPY_CTRL2_MEM_SIZE_MASK);
diff --git a/arch/gaudi2/common/hl.h b/arch/gaudi2/common/hl.h
index 2479020..d608d77 100644
--- a/arch/gaudi2/common/hl.h
+++ b/arch/gaudi2/common/hl.h
@@ -14,6 +14,6 @@
 void halt_n_reset_arc(u8 id);
 void run_arc(u8 id);
 void preboot_pcie_monitoring(void);
-void init_arc_dccm(u8 id);
+void init_arc_mem(u8 id, u64 addr, u32 size);
 
 #endif /* ARCH_GAUDI2_COMMON_HL_H_ */
